### ISSUE TITLE
Error monitoring Kubernetes pods: System.IO.IOException - The request was aborted

### ISSUE THREAD

**Issue by cdemi** [dotnet/orleans#8246]
Hi all,

We have been running Orleans 3.x on AKS for quite some time now. We are currently on 3.6.2 and we frequently have the following error logged by `Orleans.Hosting.Kubernetes.KubernetesClusterAgent`. 

We do not see any side effects and in fact Kubernetes clustering has been running well.

What is interesting to note is that we have many separate Orleans clusters on the same Kubernetes and all of them log these errors at the same time.

Since everything is working correctly I haven't investigated this problem much but they do spam quite a lot
![image](https://user-images.githubusercontent.com/8025435/210184825-ae65743f-caa2-41a3-968a-df910aca8b6a.png)

This is the full stack trace:

```
{
  "Depth": 0,
  "StackTraceString": "   at System.Net.Http.Http2Connection.ThrowRequestAborted(Exception innerException)\n   at System.Net.Http.Http2Connection.Http2Stream.CheckResponseBodyState()\n   at System.Net.Http.Http2Connection.Http2Stream.TryReadFromBuffer(Span`1 buffer, Boolean partOfSyncRead)\n   at System.Net.Http.Http2Connection.Http2Stream.ReadDataAsync(Memory`1 buffer, HttpResponseMessage responseMessage, CancellationToken cancellationToken)\n   at k8s.LineSeparatedHttpContent.CancelableStream.ReadAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)\n   at System.IO.StreamReader.ReadBufferAsync(CancellationToken cancellationToken)\n   at System.IO.StreamReader.ReadLineAsyncInternal(CancellationToken cancellationToken)\n   at k8s.Watcher`1.<>c.<CreateWatchEventEnumerator>b__21_1[TR](Task`1 t)\n   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()\n   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)\n--- End of stack trace from previous location ---\n   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)\n   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)\n--- End of stack trace from previous location ---\n   at k8s.Watcher`1.CreateWatchEventEnumerator(Func`1 streamReaderCreator, Action`1 onError, CancellationToken cancellationToken)+MoveNext()\n   at k8s.Watcher`1.CreateWatchEventEnumerator(Func`1 streamReaderCreator, Action`1 onError, CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()\n   at k8s.Watcher`1.WatcherLoop(CancellationToken cancellationToken)\n   at k8s.Watcher`1.WatcherLoop(CancellationToken cancellationToken)\n   at System.Threading.Channels.AsyncOperation`1.GetResult(Int16 token)\n   at Orleans.Hosting.Kubernetes.KubernetesClientExtensions.WatchAsync[TList,TValue](HttpOperationResponse`1 watchList, CancellationToken cancellation)+MoveNext() in /_/src/Orleans.Hosting.Kubernetes/KubernetesClientExtensions.cs:line 72\n   at Orleans.Hosting.Kubernetes.KubernetesClientExtensions.<>c__DisplayClass0_0`2.<<WatchAsync>b__4>d.MoveNext() in /_/src/Orleans.Hosting.Kubernetes/KubernetesClientExtensions.cs:line 62\n--- End of stack trace from previous location ---\n   at Orleans.Hosting.Kubernetes.KubernetesClusterAgent.MonitorKubernetesPods() in /_/src/Orleans.Hosting.Kubernetes/KubernetesClusterAgent.cs:line 275",
  "Message": "The request was aborted.",
  "HResult": -2146232800,
  "RemoteStackTraceString": null,
  "HelpURL": null,
  "RemoteStackIndex": 0,
  "ClassName": "System.IO.IOException",
  "Source": "System.Net.Http"
},
{
  "Depth": 1,
  "StackTraceString": "   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)\n   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken)\n   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)\n   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](Memory`1 buffer, CancellationToken cancellationToken)\n   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)\n   at System.Net.Http.Http2Connection.ProcessIncomingFramesAsync()",
  "Message": "Unable to read data from the transport connection: Connection reset by peer.",
  "HResult": -2146232800,
  "RemoteStackTraceString": null,
  "HelpURL": null,
  "RemoteStackIndex": 0,
  "ClassName": "System.IO.IOException",
  "Source": "System.Net.Sockets"
},
{
  "Depth": 2,
  "StackTraceString": null,
  "Message": "Connection reset by peer",
  "HResult": -2147467259,
  "RemoteStackTraceString": null,
  "HelpURL": null,
  "RemoteStackIndex": 0,
  "ClassName": "System.Net.Sockets.SocketException",
  "Source": null
}
```

It looks to me like the problem is communicating to the Kubernetes API, however since we are on AKS (with 99.95% availability for API option) I don't think there's much we can.

My main concern is that it looks like we are the only ones to get this, so it gets me to thinking that something is wrong with our setup

**Comment by cdemi** [comment]
Something interesting that I've noticed is that it happens approximately every 4 minutes and a few seconds. The Azure Load Balancer has an ‘idle timeout’ setting of 4 minutes. I tried to find if there was a way to send keep alives of less than 4 minute but I don't think I can

**Comment by lilinvictorms** [comment]
We are seeing same issue in our Orleans cluster running in AKS.

**Comment by remcoros** [comment]
Same here, running on AKS.

**Comment by cdemi** [comment]
This is fixed in Orleans 3.7.0 and up since the KubernetesClient was upgraded to v9.0.38
